includes/html/forms/application-update.inc.php

```diff
@@ -23,32 +23,33 @@
  * @author     Tony Murray <murraytony@gmail.com>
  */
 use App\Models\Application;
+use App\Models\Device;
 
 if (! Auth::user()->hasGlobalAdmin()) {
     $status = ['status' => 1, 'message' => 'You need to be admin'];
 } else {
-    $device_id = $_POST['device_id'];
+    $device = Device::find($_POST['device_id']);
     $app = $_POST['application'];
 
-    if (! isset($app) && validate_device_id($device_id) === false) {
+    if (! isset($app) && $device === null) {
         $status = ['status' => 1, 'message' => 'Error with data'];
     } else {
         $status = ['status' => 1, 'message' => 'Database update failed'];
-        $app = Application::withTrashed()->firstOrNew(['device_id' => $device_id, 'app_type' => $app]);
+        $app = Application::withTrashed()->firstOrNew(['device_id' => $device->device_id, 'app_type' => $app]);
```

Comment:
These could use the relationship.
```suggestion
        $app = $device->applications()->withTrashed()->firstOrNew(['app_type' => $app]);
```